### ISSUE TITLE
Bad file descriptor (Errno::EBADF) error when shutting down the web api webserver 

### ISSUE THREAD

**Issue by purbon** [elastic/logstash#4674]
Sometimes when shutting down logstash puma shows this error:

```
Error in reactor loop escaped: Bad file descriptor - Bad file descriptor (Errno::EBADF)
org/jruby/RubyIO.java:3681:in `select'
/Users/purbon/work/logstash/vendor/bundle/jruby/1.9/gems/puma-2.16.0-java/lib/puma/reactor.rb:29:in `run_internal'
/Users/purbon/work/logstash/vendor/bundle/jruby/1.9/gems/puma-2.16.0-java/lib/puma/reactor.rb:138:in `run_in_thread'
```

is not something causing problem in runtime, but should be investigated and avoided.

**Comment by purbon** [comment]
For now I'm unable to reproduce this, something might have changed internally that made this error go away. @ph @suyograo If you see this error, please let me know, but for now I think is just gone :+1:

**Comment by purbon** [comment]
I just deleted my previous comment, I can see this from running full integrated test:core as of #4667

**Comment by purbon** [comment]
A fix for this goes into #4782
